store/cachekv/store.go

```diff
@@ -186,24 +126,28 @@ func (store *Store) Write() {
 			continue
 		}
 
-		cacheValue, _ := store.cache.Get(key)
-		if cacheValue.Value() != nil {
+		cacheValue, ok := store.cache.Load(key)
+		if ok && cacheValue.(*types.CValue).Value() != nil {
 			// It already exists in the parent, hence delete it.
-			store.parent.Set([]byte(key), cacheValue.Value())
+			store.parent.Set([]byte(key), cacheValue.(*types.CValue).Value())
 		}
 	}
 
 	// Clear the cache using the map clearing idiom
 	// and not allocating fresh objects.
 	// Please see https://bencher.orijtech.com/perfclinic/mapclearing/
-	store.cache.DeleteAll()
+	store.cache.Range(func(key, value any) bool {
```

Comment:
I've thought about that, but I think it could be risky, I'm not sure why initially we are not doing that tbh, so better to just keep the same logic first.